src/construction.cpp

```diff
@@ -851,16 +876,27 @@ void complete_construction()
 
         award_xp( *elem );
     }
-
-    for( const auto &it : built.requirements->get_components() ) {
-        u.consume_items( it, 1, is_crafting_component );
+    g->m.disarm_trap( terp );
+    g->m.partial_con_remove( terp );
+    // Move any items that have found their way onto the construction site.
+    std::vector<tripoint> dump_spots;
+    for( const auto pt : g->m.points_in_radius( terp, 1 ) ) {
```

Comment:
Pretty sure this includes `terp`, where dumping items would definitely be counterproductive.